confidant/routes/credentials.py

```diff
@@ -765,22 +768,34 @@ def update_credential(id):
     if _cred.data_type != 'credential':
         msg = 'id provided is not a credential.'
         return jsonify({'error': msg}), 400
+
     data = request.get_json()
-    update = {}
-    revision = credentialmanager.get_latest_credential_revision(
-        id,
-        _cred.revision
-    )
-    update['name'] = data.get('name', _cred.name)
+    if not isinstance(data.get('metadata', {}), dict):
+        return jsonify({'error': 'metadata must be a dict'}), 400
+
+    update = {
+        'name': data.get('name', _cred.name),
+        'last_rotation_date': _cred.last_rotation_date,
+        'credential_pairs': _cred.credential_pairs,
```

Comment:
changing to use the encrypted string rather than the decryped dict as Ciphermanager.encrypt expects a string

Reply:
That's going to be an encrypted string. You're going to double encrypt here.

Reply:
you want: `json.dumps(_cred.decrypted_credential_pairs)`

Reply:
alternatively, you can keep the encrypted value here, and move the part that does encryption into the if section and only apply it to what's in the data field

Reply:
Ah you're right